File: 9/717-1-bit-and-2-bit-characters/main.go
package main

import "fmt"

func isOneBitCharacter(bits []int) bool {
	s := 0
	for s < len(bits)-1 {
		if bits[s] == 0 {
			s++
		} else {
			s += 2
		}
	}
	return s == len(bits)-1
}

func main() {
	fmt.Println("a")
}

File: 9/717-1-bit-and-2-bit-characters/main.rs
impl Solution {
  pub fn is_one_bit_character(bits: Vec<i32>) -> bool {
    let mut idx: usize = 0;
    while idx < bits.len() - 1 {
      idx += bits[idx] as usize + 1;
    }
    idx == bits.len() - 1
  }
}
